bundles/org.smarthomej.binding.androiddebugbridge/src/main/java/org/smarthomej/binding/androiddebugbridge/internal/AndroidDebugBridgeDevice.java

```diff
@@ -254,16 +278,18 @@ public void connect() throws AndroidDebugBridgeDeviceException, InterruptedExcep
             if (e.getMessage().equals("Socket closed")) {
                 // Connection aborted by us
                 throw new InterruptedException();
+            } else if (e.getMessage().equals("No route to host (Host unreachable)")) {
+                throw new TimeoutException();
```

Comment:
Are these exceptions treted differently? Otherwise it's usually better to stick with one exception type.

Reply:
yes because in the upper method it use the exception to decide what to show on the frontend: offline or err:communication